plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java

```diff
@@ -146,19 +147,23 @@ private static RemoteIterator<TrinoFileStatus> createListingRemoteIterator(Trino
     @Override
     public void invalidate(Location location)
     {
-        cache.invalidate(location);
+        log.debug("Invalidating cache for location: %s", location);
+        cache.asMap().keySet().removeIf(key -> key.location().equals(location));
     }
 
     @Override
     public void invalidate(Table table)
     {
         if (isCacheEnabledFor(table.getSchemaTableName()) && isLocationPresent(table.getStorage())) {
```

Comment:
The `isLocationPresent` check here seems suspicious (for both `invalidate(Table)` and `invalidate(Partition)`). It would be good to look back at usage sites and see whether this can be simplified to a single `invalidate(Table, Location)` method that always has both arguments- because skipping an invalidation due to a missing location seems like a bug waiting to happen.

Reply:
cool yeah this seems totally doable, most of the changes are from the tests.  I can make this work.  Thanks for the heads up :)

Reply:
i added the table parameter for this